src/Microsoft.OpenApi/Models/OpenApiCallback.cs

```diff
@@ -45,10 +45,10 @@ public class OpenApiCallback : IOpenApiSerializable, IOpenApiReferenceable, IOpe
         /// </summary>
         public OpenApiCallback(OpenApiCallback callback)
         {
-            PathItems = new(callback.PathItems);
-            UnresolvedReference = callback.UnresolvedReference;
-            Reference = new(callback.Reference);
-            Extensions = new Dictionary<string, IOpenApiExtension>(callback.Extensions);
+            PathItems = new(callback?.PathItems);
+            UnresolvedReference = callback?.UnresolvedReference ?? false;
+            Reference = new(callback?.Reference);
+            Extensions = callback?.Extensions != null ? new Dictionary<string, IOpenApiExtension>(callback?.Extensions) : callback?.Extensions;
```

Comment:
```suggestion
            Extensions = callback?.Extensions == null ? null : new Dictionary<string, IOpenApiExtension>(callback.Extensions);
```

(and repeat for collections)